### PR TITLE
Write bullet point 2 "author shares the code"

### PR DESCRIPTION
**Pull request by tlestang**
Going from bullet points to prose. Main points are:

- We recommend sharing the code before hand
- It's important to attach local context to the code, to help reviewers form constructive feedback (and make their life easier)

There is also a short discussion about how to share the code in practice.

### CODE REVIEW

**Comment by NickleDave** [comment]
@tlestang I fixed some typos in one commit, and then revised the language a bit in a separate commit.

My goal with the second commit was just to tone down a bit terms like "always" and "it's not enough" that just sounded a tad bit aggressive.

If you don't like that we can just drop that commit -- just adding my take as it hits me.

**Comment by NickleDave** [comment]
@tlestang I fixed some typos in one commit, and then revised the language a bit in a separate commit.

My goal with the second commit was just to tone down a bit terms like "always" and "it's not enough" that just sounded a tad bit aggressive.

If you don't like that we can just drop that commit -- just adding my take as it hits me.

File: site/content/recipes/perform_code_review.md [hunk 1]
```diff
@@ -16,20 +16,22 @@ explore the code at their own pace and form independent
...
+will also want to help the reviewer understand
+how the particular snippet you select fits into the overall code.
+This helps the reviewer provide relevant comments
+and constructive feedback.
+As an illustration, picture the case where the code under review consists of
+a couple methods belonging to a class. The methods were implemented for a
+specific purpose, and they might interact in a specific way.
+The methods also play particular role within the class
+itself, a class that is designed in its own way, to fit its own purpose.
+The purpose, the design and the implementation of the methods and their
```
Comment (tlestang) [inline site/content/recipes/perform_code_review.md:29]: I like it!

File: site/content/recipes/perform_code_review.md [hunk 2]
```diff
@@ -10,19 +10,45 @@
...
+the bigger logic and explicit the relationships with other parts that
+use it or are used by it.
+
+> **How to share code?**
+> Ideally, revisions of your research software should be available from
+> development platforms like GitHub, GitLab or Bitbucket. Simply give
+> the reviewer a permalink (see figure ??) or a commit SHA-1 and
+> relevant line numbers. This is the preferred option as it gives
+> reviewers the opporunity to navigate the whole codebase. Another
+> option is [GitHub gists](https://gist.github.com/). If you send share
```
Comment (bielsnohr) [inline site/content/recipes/perform_code_review.md:46]:
```suggestion
> option is [GitHub gists](https://gist.github.com/). If you send
```

File: site/content/recipes/perform_code_review.md [hunk 3]
```diff
@@ -10,19 +10,45 @@
...
+> your code by email, make sure to share the relevant files and clearly
+> identify the version you are sharing.
+
+![To get a link to a particular line (permalink) on GitHub, click on the line number on the left hand side](/dev-review/images/github_permalink.png "To get a link to a particular line (permalink) on GitHub, click on the line number on the left hand sise. Most development platforms can generate links to a particular line, for a particular revision.")
```
Comment (bielsnohr) [inline site/content/recipes/perform_code_review.md:50]: This is a great idea! On a pedantic kerning note, we should agree about line length. But again, an issue for another day.

**Review by bielsnohr — APPROVED** [review]
Overall, this is a very good first pass. I have made one minor typo edit that I will add as a commit, and then merge. The other points are for us to address in the future, and I will make into issues.

**Pull request merged**